third_party/validator/folder_iam.go

```diff
@@ -0,0 +1,48 @@
+package google
+
+import "fmt"
+
+func GetFolderIamPolicyCaiObject(d TerraformResourceData, config *Config) (Asset, error) {
+	return newFolderIamAsset(d, config, expandIamPolicyBindings)
+}
+
+func GetFolderIamBindingCaiObject(d TerraformResourceData, config *Config) (Asset, error) {
+	return newFolderIamAsset(d, config, expandIamRoleBindings)
+}
+
+func GetFolderIamMemberCaiObject(d TerraformResourceData, config *Config) (Asset, error) {
+	return newFolderIamAsset(d, config, expandIamMemberBindings)
+}
+
+func MergeFolderIamPolicy(existing, incoming Asset) Asset {
+	existing.IAMPolicy = incoming.IAMPolicy
+	return existing
+}
+
+func MergeFolderIamBinding(existing, incoming Asset) Asset {
+	return mergeIamAssets(existing, incoming, mergeAuthoritativeBindings)
+}
+
+func MergeFolderIamMember(existing, incoming Asset) Asset {
+	return mergeIamAssets(existing, incoming, mergeAdditiveBindings)
+}
+
+func newFolderIamAsset(
+	d TerraformResourceData,
+	config *Config,
+	expandBindings func(d TerraformResourceData) ([]IAMBinding, error),
+) (Asset, error) {
+	bindings, err := expandBindings(d)
+	if err != nil {
+		return Asset{}, fmt.Errorf("expanding bindings: %v", err)
```

Comment:
Usually in go we return `nil, error` instead of instantiating an empty struct. I see it as an extra layer of protection to prevent somebody from accidentally forgetting to check errors and unintentionally picking up a valid looking object.

Reply:
This is to stay consistent with the other function signatures. If we prefer to switch all funcs over to returning a pointer (`*Asset`) I am ok with that as well, but think that should be in another PR.